### ISSUE TITLE
Add UUID version 6 and 7 to stdlib (Lib/uuid.py) once RFC 4122 becomes updated

### ISSUE THREAD

**Issue by jgehrcke** [python/cpython#102461]
An update to RFC 4122 is well underway: https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/ 
The last version (as of the time of writing this) is from 16 February 2023.

Editorial work seems to be happening here: https://github.com/ietf-wg-uuidrev/rfc4122bis
High-level updates are communicated here: https://github.com/uuid6/uuid6-ietf-draft/issues/122

Among others, this updated standard introduces UUID versions 6 and 7. The Internet community seems to be excited about those. I trust that it's rather uncontroversial that at least those two should be added to Python's standard library once the standard is complete. But of course we should collect opinions about that idea here.

Prior art:
- Someone submitted a patch on Nov 28, 2021 without further discussion: https://github.com/python/cpython/pull/29824
- There is https://github.com/stevesimmons/uuid7, published as https://pypi.org/project/uuid7
- There is https://github.com/oittaa/uuid6-python, published as https://pypi.org/project/uuid6

**Comment by terryjreedy** [comment]
uuid.py is actively maintained, but no one has claimed it at https://devguide.python.org/core-developers/experts/.

@jgehrcke Are you in a position to submit a PR at the appropriate time?

@tiran Does your security interest include this module?  You patched it last July.

**Comment by jgehrcke** [comment]
@terryjreedy I think so, and I'd love to! Maybe @orsenthil can also help me with that (hello :wave:).

**Comment by sergeyprokhorenko** [comment]
I recommend using the following implementations as a basis:
[JavaScript](https://github.com/LiosK/uuidv7)
[Rust](https://crates.io/crates/uuid7)
[PostgreSQL](https://commitfest.postgresql.org/46/4388/)
[Go language](https://pkg.go.dev/github.com/gofrs/uuid#NewV7)

**Comment by hugovk** [comment]
For reference, see discussion at https://discuss.python.org/t/add-uuid7-in-uuid-module-in-standard-library/44390.

**Comment by hugovk** [comment]
Let's close this as a duplicate of the older https://github.com/python/cpython/issues/102461, which also mentions UUIDv8.
